hadoop-ozone/integration-test-s3/src/test/java/org/apache/hadoop/ozone/s3/ProxyServer.java

```diff
@@ -0,0 +1,251 @@
+      HttpURLConnection conn = (HttpURLConnection) new URL(target).openConnection();
+      String requestMethod = exchange.getRequestMethod();
+      conn.setRequestMethod(requestMethod);
+      conn.setConnectTimeout(CONNECT_TIMEOUT_MS);
+      conn.setReadTimeout(READ_TIMEOUT_MS);
+      return conn;
+    }
+
+    private void copyRequestBody(HttpExchange exchange, HttpURLConnection conn) throws IOException {
+      String method = exchange.getRequestMethod().toLowerCase();
+      if (method.equals("post") || method.equals("put") || method.equals("patch") || method.equals("delete")) {
+        String contentLength = exchange.getRequestHeaders().getFirst("Content-Length");
+        conn.setDoOutput(true);
+        if (StringUtils.isNotBlank(contentLength)) {
+          conn.setFixedLengthStreamingMode(Long.parseLong(contentLength));
+        } else {
+          conn.setChunkedStreamingMode(0);
+        }
+        try (InputStream reqBody = exchange.getRequestBody();
+             OutputStream os = conn.getOutputStream()) {
+          IOUtils.copy(reqBody, os);
+        }
+      }
+    }
+
+    private void copyRequestHeaders(HttpExchange exchange, HttpURLConnection conn) {
+      for (String headerName : exchange.getRequestHeaders().keySet()) {
+        if (isHopByHopHeader(headerName)) {
+          continue;
+        }
+
+        // Skip Expect header to avoid 100-continue issue
+        if ("Expect".equalsIgnoreCase(headerName)) {
+          continue;
+        }
+
+        List<String> headerValues = exchange.getRequestHeaders().get(headerName);
+        if (headerValues == null || headerValues.isEmpty()) {
+          continue;
+        }
+
+        for (String value : headerValues) {
+          // Convert header names to lowercase because HttpServer capitalizes the first letter,
+          // but s3g expects lowercase header names.
+          conn.addRequestProperty(headerName.toLowerCase(), value);
+        }
+      }
+    }
+  }
+
+  private void copyResponseHeaders(HttpExchange exchange, HttpURLConnection conn) {
+    for (String headerName : conn.getHeaderFields().keySet()) {
+      if (headerName == null || isHopByHopHeader(headerName)) {
+        continue;
+      }
+
+      List<String> headerValues = conn.getHeaderFields().get(headerName);
+      if (headerValues == null) {
+        continue;
+      }
+
+      for (String headerValue : headerValues) {
+        exchange.getResponseHeaders().add(headerName, headerValue);
+      }
+    }
+  }
+
+  private boolean isHopByHopHeader(String header) {
+    String lowerName = header.toLowerCase();
+    return "connection".equals(lowerName) || "keep-alive".equals(lowerName)
+        || "proxy-authenticate".equals(lowerName) || "proxy-authorization".equals(lowerName)
+        || "te".equals(lowerName) || "trailers".equals(lowerName)
+        || "transfer-encoding".equals(lowerName) || "upgrade".equals(lowerName);
```

Comment:
We had a previous issues where `transfer-encoding` and `expect` was included in the signed headers (atlhough it is a hop-by-hop headers), but seems to have been dropped by the proxy. Triggering the following exceptions.

```
InternalError: Header expect not present in request but requested to be signed.\n\tstatus code: 500, request id: 28c2d896-d04c-4068-b854-9b036c03671f, host id: mQxBdH9Q8loGf 
```

However, if it's passing on the current AWS SDK tests, I'm ok with dropping these request headers.

Nit: Can add some documentation link regarding hop-by-hop headers.